src/annotator/guest.ts

```diff
@@ -706,7 +707,16 @@ export class Guest extends TinyEmitter implements Annotator, Destroyable {
    * Show or hide the adder toolbar when the selection changes.
    */
   _onSelection(range: Range) {
-    if (!this._integration.canAnnotate(range)) {
+    let textRange: Range;
+    try {
+      textRange = TextRange.fromRange(range).toRange();
+      textRange = trimRange(textRange);
+    } catch {
+      this._onClearSelection();
```

Comment:
I think it would be useful to comment here when failure is expected to happen. Also, should we be checking for specific failures here (eg. `RangeError`) and allowing unexpected failures to propagate?